#!/usr/bin/env python
PACKAGE = "sensor_fusion"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("downsamle_seikowave", bool_t, 0, "Decision to downsample Seikowave", False)
gen.add("pc_keep_num", double_t,    0, "Seikowave point keep frequency", 3.0,  0, 100)
gen.add("concatenate_output", bool_t,    0, "To concatenate the pointclouds", True)
gen.add("republish_pointclouds", bool_t,    0, "To re-publish pointclouds or not", True)

gen.add("x_motion", double_t,    0, "Move pointcloud in x", 0.0, -0.5, 0.5)
gen.add("y_motion", double_t,    0, "Move pointcloud in y", 0.0, -0.5, 0.5)
gen.add("z_motion", double_t,    0, "Move pointcloud in z", 0.0, -0.5, 0.5)

exit(gen.generate(PACKAGE, "sensor_fusion", "fusion"))
